src/md5/md5.cc

```diff
@@ -119,11 +119,11 @@ void MD5::update ( const uint1 *input, uint4 input_length) {
 
 void MD5::update(FILE *file){
 
-  unsigned char buffer[1024];
-  int len;
+  unsigned char fileBuffer[1024];
+  size_t len;
```

Comment:
Why not `uint4`/`uint32_t`? Then you wouldn't need the `static_cast`.

Reply:
Because ``fread`` returns a ``size_t`` which may be one of these but also can be ``uint64_t`` or whatever ``typedef /*implementation-defined*/ size_t;`` is in the used implementation. So in these locations i used the types native to the functions that write them and cast them to the types the functions need that read them.